pjsip/include/pjsua2/siptypes.hpp

```diff
@@ -307,6 +307,72 @@ struct TlsConfig : public PersistentObject
 };
 
 
+/**
+ * Socket option parameters, to be specified in TransportConfig.
+ */
+struct SockOptParams : public PersistentObject
+{
+    /**
```

Comment:
if I remember correctly, this kind of nested structure needs to be flattened. I don't remember the exact reason, but maybe it's SWIG?

Also, just curious, can this work in Java?

Reply:
Haven't tested with SWIG, will do it later.